### PR TITLE
LPS-124110 Add external reference code to BlogsEntry

### PR DESCRIPTION
**Pull request by LuismiBarcos**
This pull is the first part of the feature described in [this ticket](https://issues.liferay.com/browse/LPS-124110)
The external reference code (also called ERC) is a unique id to identify a specific resource. It could be scoped by group or by company.

In the PR, it added support for external reference code in BlogsEntry services to be able to add BlogsEntries with a specific external reference code. In case that the user does not provide the ERC, it will be assigned by default the id of the resource.

### CODE REVIEW

File: modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -270,65 +270,6 @@ public BlogsEntry addEntry(
...
-
-	/**
-	 * @deprecated As of Cavanaugh (7.4.x), replaced by {@link
-	 * #addEntry(String, long, String, String, String, String, String, int, int,
-	 * int, int, int, boolean, boolean, String[], String, ImageSelector,
-	 * ImageSelector, ServiceContext)}
-	 */
-	@Deprecated
-	@Override
-	public BlogsEntry addEntry(
```
Comment (javierdearcos) [inline modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java]: Did you check the usage of this method? I didn't find any code related to the replacement of this method calls in the PR changes. Maybe it was unused?

File: modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BaseBatchEngineTaskExecutorTest.java [hunk 2]
```diff
@@ -176,7 +176,8 @@ public void createItem(
 				blogPosting.getDatePublished());
 
 			_blogsEntryService.addEntry(
```
Comment (javierdearcos) [inline modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BaseBatchEngineTaskExecutorTest.java:178]: Once we are going to remove old methods, I have doubts if it is better to include the externalReferenceCode argument in every addEntry signatures and add the "null" value in the usage of the methods instead having here methods with and without externelReferenceCode

File: modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java [hunk 3]
```diff
@@ -244,8 +244,8 @@ public BlogsEntry addEntry(
 		throws PortalException {
 
 		return blogsEntryLocalService.addEntry(
-			userId, title, subtitle, StringPool.BLANK, description, content,
-			displayDate, allowPingbacks, allowTrackbacks, trackbacks,
+			null, userId, title, subtitle, StringPool.BLANK, description,
```
Comment (javierdearcos) [inline modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java:247]: Once we are going to remove old methods, I have doubts if it is better to include the externalReferenceCode argument in every addEntry signatures and add the "null" value in the usage of the methods instead having here methods with and without externelReferenceCode

**Review by javierdearcos — APPROVED** [review]
Double check if we want to leave methods with and without externalReferenceCode in the BlogsEntryLocalService interface, or add externalReferenceCode to every addEntry method

**Review by nhpatt — APPROVED** [review]

**Comment by LuismiBarcos** [comment]
> we want to leave methods with and without externalReferenceCode in the BlogsEntryLocalService interface, or add externalReferenceCode to every addEntry method

My thought about that is that as every method will end calling the method with the ERC I think it is not necessary to force the user to know the ERC. In addition, the deprecation and deletion of the methods will be a problem in the upgrade process ( I know that the new plans for the upgrade will avoid these problems in migrations but I am not sure if we will see that in this release 😁 )

Because of that, I did not add the ERC to every method. Also, the entities that are actually in master have more than one creation method without ERC, so we will introduce some inconsistency 😁 

I don't know what do you think about this.

**Comment by nhpatt** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
Fix the SF, rebase and resend.

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:stop

**Comment by LuismiBarcos** [comment]
I stop the CI to check that in this implementation I am not including [this bug](https://issues.liferay.com/browse/LPS-134933)

**Comment by LuismiBarcos** [comment]
Rebase and fix the problem with an empty string in ERC. I forward the pull directly

**Comment by LuismiBarcos** [comment]
ci:forward

**Pull request closed without merging**